tests/raftstore/test_multi.rs

```diff
@@ -267,3 +268,88 @@ fn test_multi_server_random_restart() {
     let mut cluster = new_server_cluster(0, count);
     test_multi_random_restart(&mut cluster, count, 10);
 }
+
+fn test_leader_change_with_uncommitted_log<T: Simulator>(cluster: &mut Cluster<T>) {
+    cluster.cfg.store_cfg.raft_election_timeout_ticks = 50;
+    // disable compact log to make test more stable.
+    cluster.cfg.store_cfg.raft_log_gc_threshold = 1000;
+    // We use three peers([1, 2, 3]) for this test.
+    cluster.run();
+
+    sleep_ms(500);
```

Comment:
Because election timeout has been enlarged, so here wait a little time in case CI is very slow.